### PR TITLE
Restructure Settings, enable dark theme - Fix #840

### LINKED ISSUE #840: Restructure settings area, avoid popups, ensure same colors
The settings area has grown organically. It is time to align it closer to other existing services.

* Please re-arrange items in the settings tab following the Twitter settings page structure, e.g. add a tab "Password" and a tab "Account" that shows the email of the user.
* Ensure all buttons use the same blue color, e.g. in theme tab
* If "Dark" theme or other theme options are chosen, please also use it in the settings area.

screenshot from 2017-09-28 10-36-32

### LINKED ISSUE #841: Bugs in Settings after saving the state
**Actual Behaviour**

There are two main issues with the save button and the settings page:
1. On pressing save, the user is not redirected to the home page
2. The option being changed, although updated internally, is not reflected when I hit save. For example. if I change the theme to "dark", even after pressing the save button, the "light" radio button is selected.  This issue of the old state being shown happens for ALL the tabs in Settings, not just the theme tab. The only way for the user to actually verify that the change has taken place is to go to the homepage manually. Sometimes, in some cases, refreshing the Settings page, shows the new value but not by default.

**Expected Behaviour**

1. The user should be redirected to the homepage.
2. The changed state should be reflected on clicking save.

**Steps to reproduce it**

1. Go to Settings
2. Click on say, the Theme tab or the ChatApp Settings Tab.
3. Change the value and click Save.
4. The user is not redirected to the homepage and the old values are being displayed.

**Screenshots of the issue**

Initial State:
before

Just before clicking the Save button:
middle

After clicking save:
final

**Would you like to work on the issue?**

Yes, I would like to work on the issue, but since it is a relatively big issue, I would be requiring some guidance. Thanks :)

### PR DESCRIPTION
**Pull request by hkedia321**
Fixes issue #840 

Changes:
- Fix bug in saving dark theme while signed in - fixes #841 
- Show User's email address
- Remove unnecessary popups in Settings page
- Disable refreshing after changing settings
- Save changes only for the changed settings of the current tab
- Enable dark theme in Settings page
- Integrate Mobile page

Demo Link: [clammy-park.surge.sh](http://clammy-park.surge.sh)

Screenshots for the change: 
image
image

### CODE REVIEW

**Comment by hkedia321** [comment]
@isuruAb Now changing the settings is working. Please check. Let me know if any changes are required.

File: src/components/ChatApp/Settings/Settings.react.js [hunk 1]
```diff
@@ -227,7 +249,7 @@ class Settings extends Component {
 	  Actions.resetVoice();
 	}
 	this.props.history.push(`/settings?tab=${this.state.selectedSetting}`);
-	window.location.reload();
+	// window.location.reload();
```
Comment (isuruAb) [inline src/components/ChatApp/Settings/Settings.react.js]: please remove unnecessary  comments

File: src/actions/API.actions.js [hunk 2]
```diff
@@ -365,6 +367,10 @@ export function getSettings(){
       timeout: 3000,
       async: false,
       success: function (response) {
+          console.log(response)
```
Comment (isuruAb) [inline src/actions/API.actions.js]: please remove unwanted logs 👍

**Comment by hkedia321** [comment]
@isuruAb @madhavrathi I have removed the unwanted logs and comments. :+1:  Thanks!

**Comment by hkedia321** [comment]
@mariobehling I have changed the required text and updated the surge (link)[http://breakable-kittens.surge.sh]. Please let me know if any more changes are required. :+1:

**Comment by isuruAb** [comment]
@hkedia321  please resolve conflicts

**Comment by hkedia321** [comment]
@isuruAb @mariobehling I have resolved the merge conflicts, and have updated the screenshots and surge [link](http://untidy-roof.surge.sh/). Please have a look.

**Comment by mariobehling** [comment]
Good job! Unfortunately there was another PR in the queue that resulted in another conflict. Please help. Thanks.

**Comment by isuruAb** [comment]
@hkedia321  please resolve conflicts 👍

**Comment by hkedia321** [comment]
@mariobehling @isuruAb Resolved the merge conflicts! Here is the updated surge: [clammy-park.surge.sh](http://clammy-park.surge.sh). Please review.

**Comment by mariobehling** [comment]
Thanks! Good job.

**Pull request merged**